sentry_streams/tests/adapters/arroyo/test_broadcaster.py

```diff
@@ -0,0 +1,114 @@
+from datetime import datetime
+from typing import Any
+from unittest import mock
+from unittest.mock import call
+
+from arroyo.processing.strategies.abstract import ProcessingStrategy
+from arroyo.types import FilteredPayload, Message, Partition, Topic, Value
+
+from sentry_streams.adapters.arroyo.broadcaster import Broadcaster
+from sentry_streams.adapters.arroyo.routes import Route, RoutedValue
+
+
+# TODO: move make_value_msg into some kind of test utils folder as it's shared with other tests
+def make_value_msg(payload: Any, route: Route, offset: int) -> Message[Any]:
+    """
+    Makes a message containing a Value based on the offset passed.
+    Useful if a step you're testing always transforms a Message payload into a Value,
+    or if you need an emtpy comittable/timestamp for whatever reason (BrokerValue doesn't support that).
+    """
+    timestamp = datetime(2025, 1, 1, 12, 0)
+    if isinstance(payload, FilteredPayload):
+        return Message(
+            Value(
+                payload=payload,
+                committable={Partition(Topic("test_topic"), 0): offset},
+                timestamp=timestamp,
+            )
+        )
+    else:
+        return Message(
+            Value(
+                payload=RoutedValue(route=route, payload=payload),
+                committable={Partition(Topic("test_topic"), 0): offset},
+                timestamp=timestamp,
+            )
+        )
+
+
+def test_submit() -> None:
```

Comment:
Please test the negative cases where you send: FilteredMessage and messages for another route.
Also please test the case where the downstream step raises MessageRejected on the second route after succeeding the first time. Ensure the message is forwarded only once per route.

When writing unit tests it is all the more important to test teh corner cases as they are the easier ones to break without noticing.